js/script.js

```diff
@@ -0,0 +1,262 @@
+    var computerScoreSpan = document.getElementById("computer-score");
+    var output = document.getElementById("output");
+
+    var modals = document.querySelectorAll('.modal');
+    var modalNewGame = document.getElementById("modal-newGame")
+    var modalScore = document.getElementById("modal-score");
+    var modalOutput = document.getElementById("modal-score__output");
+    var closeButtons = document.querySelectorAll('.modal .close');
+    var modalTableBody = document.getElementById('game-table__body');
+
+    var params = {
+        pointsToWin: null,
+        roundCount: 0,
+        userScore: 0,
+        computerScore: 0,
+        playerChoice: undefined,
+        computerChoice: undefined,
+        roundResult: undefined,
+        progress: []
+    }
+
+    var gameData = {
+        roundNmbr: undefined,
+        playerMv: undefined,
+        computerMv: undefined,
+        roundRslt: undefined,
+        gameScr: undefined
+    }
+
+    var isInputValid = function (pointsToWin) {
+        if (!params.pointsToWin || isNaN(params.pointsToWin) || params.pointsToWin < 1 || params.pointsToWin > 10) {
+            return alert('Please enter a number between 1 and 10.');
+        }
+        return true;
+    }
+
+    var newGame = function () {
+        params.pointsToWin = roundLimit.value;
+        if (isInputValid(params.pointsToWin)) {
+            document.getElementById('newGame-btn').style.display = "none";
+            document.getElementById('scoreboard').style.display = "block";
+            document.getElementById('memo').style.display = "block";
+            for (var i=0; i < buttons.length; i++) {
+                buttons[i].style.display = "inline-block";
+            }
+            activateButtons();
+            params.roundCount = 0;
+            params.progress = [];
+            params.userScore = 0;
+            userScoreSpan.innerHTML = params.userScore;
+            params.computerScore = 0;
+            computerScoreSpan.innerHTML = params.computerScore;
+            while (output.hasChildNodes ()) {
+                output.removeChild(output.firstChild);
+            }
+        }
+    }
+    
+    var isGameFinished = function(score) {
+        if (score >= params.pointsToWin) {
+            return true;
+        }
+        return false;
+    }
+
+    var disableButtons = function() {
+    	for (i = 0; i < buttons.length; i++) {
+            buttons[i].disabled = true;
+        }
+    }
+    
+    var activateButtons = function() {
+        for (i = 0; i < buttons.length; i++) {
+            buttons[i].disabled = false;
+        }
+    }
+
+    var playerMove = function(playerPick) {
+        params.roundCount = ++params.roundCount;
+        params.playerChoice = playerPick;
+        switch (playerPick + opponentPick()) {
+            case "rockrock":
+                output.innerHTML = ('You both chose ROCK. It is a tie.');
+                params.computerChoice = 'rock';
```

Comment:
prepare one more function to prevent repeating code

```
var afterChooseAction = function (msg, computerChoice, roundResult) {
  output.innerHTML = msg;
  params = Object.assign(params, {
    computerChoice: computerChoice,
    roundResult: roundResult,
  });
}
```